holoviews/plotting/mpl/element.py

```diff
@@ -658,6 +652,14 @@ def _norm_kwargs(self, element, ranges, opts, vdim, prefix=''):
                     alpha = int(color[-2:], 16)/255.
                     color = color[:-2]
                 colors[k] = {'color': color, 'alpha': alpha}
+
+        if isinstance(cmap, dict):
+            factors = np.unique(values)
+            palette = [cmap.get(f, colors.get('NaN', {'color': '#8b8b8b'})['color'])
```

Comment:
What is this magic value '#8b8b8b'? I'm guessing it is some grayish color...

Reply:
Right, can probably specify that somewhere else.

Reply:
Is that a `default_nan_color` or `color_for_missing_values`?

Reply:
It's really ``color_for_values_which_have_not_been_assigned_a_explicit_color``.